src/anyvlm/utils/types.py

```diff
@@ -0,0 +1,74 @@
+"""Provide helpful type definitions, references, and type-based operations."""
+
+from enum import Enum, StrEnum
+from typing import Annotated
+
+from pydantic import BeforeValidator, StringConstraints
+
+
+class EndpointTag(str, Enum):
+    """Define tag names for endpoints."""
+
+    META = "Meta"
+    SEARCH = "Search"
+
+
+class GrcAssemblyId(StrEnum):
+    """Supported GRC assembly identifiers"""
+
+    GRCH37 = "GRCh37"
+    GRCH38 = "GRCh38"
+
+
+class UscsAssemblyBuild(StrEnum):
+    """Supported USCS assembly builds"""
+
+    HG38 = "hg38"
+    HG19 = "hg19"
+
+
+GenomicSequence = Annotated[
+    str,
+    BeforeValidator(str.upper),
+    StringConstraints(pattern=r"^[AGCT]*$"),
+]
+
+
+def is_valid_chromosome_name(chromosome_name: str) -> bool:
+    """Checks whether or not a provided chromosome name is valid.
+
+    :param chromosome_name: The chromosome name to validate.
+    :return: `True` if the chromosome name is a number between 1-22, or the values "X", "Y", or "MT"; else `False`.
+    """
+    min_chromosome_number = 1
+    max_chromosome_number = 22
+    try:
+        return (
+            chromosome_name in {"X", "Y", "MT"}
+            or min_chromosome_number <= int(chromosome_name) <= max_chromosome_number
+        )
+    except ValueError:
+        return False
+
+
+def _normalize_chromosome_name(chromosome_name: str) -> str:
+    """Normalize a chromosome name. Input must be a string consisting of either a number between 1-22,
+    or 'X' or 'Y'; optionally prefixed with 'chr'.
+
+    :param chromosome_name: The name of the chromosome to normalize, following the rules stated above.
+    :return: The chromosome name, stripped of it's 'chr' prefix if it was added
+    """
+    # strip the 'chr' prefix if it was included
+    chromosome_name = (
+        chromosome_name[3:]
+        if chromosome_name.lower().startswith("chr")
+        else chromosome_name
+    ).upper()
+
+    if is_valid_chromosome_name(chromosome_name):
+        return chromosome_name
+    error_message = "Invalid chromosome. Must be either a number between 1-22, or the values 'X', 'Y', or 'MT'; with optional 'chr' prefix."
+    raise ValueError(error_message)
```

Comment:
is there a reason this is broken out to another function? it means you have to catch a valueerror above, turn it into a `False`, and then turn that `False` back into a `ValueError` here.

```suggestion
    if chromosome_name in {"X", "Y", "MT"} or 1 <= int(chromosome_name) <= 22:
        return True
    error_message = "Invalid chromosome. Must be either a number between 1-22, or the values 'X', 'Y', or 'MT'; with optional 'chr' prefix."    
    raise ValueError(error_message)
```